.github/workflows/eco-branch-images.yml

```diff
@@ -0,0 +1,88 @@
+name: Eco branch images
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  UPDATER_IMAGE: "dependabot/updater-core"
+  UPDATER_IMAGE_MIRROR: "ghcr.io/dependabot/dependabot-updater-gomod"
+  DOCKERFILE: "Dockerfile.go_modules"
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - synchronize
+  pull_request_review:
+    types:
+      - submitted
+
+jobs:
+  push-updater-images:
+    name: Deploy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        suite:
+          - { name: go, ecosystem: gomod }
+          - { name: python, ecosystem: pip }
+    permissions:
+      contents: read
+      packages: write
+    env:
+      TAG: ${{ github.sha }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+        # sets DECISION to the PR's review decision, handling the push-after-approval case
+      - name: Check if pull request is approved
+        run: |
+          DECISION=$(gh pr view ${{ github.event.pull_request.number }} --json reviewDecision,state -t '{{.reviewDecision}}:{{.state}}')
+          echo "Review decision is: $DECISION"
+          echo "DECISION=$DECISION" >> $GITHUB_ENV
+
+      # overwrite the previous result if this is a fork since forks can't publish to GHCR
+      - name: Skip forks
+        if: github.event.pull_request.head.repo.fork
+        run: echo "DECISION=FORK" >> $GITHUB_ENV
+
+      - name: Build updater core image
+        env:
+          DOCKER_BUILDKIT: 1
+        run: |
+          docker build \
+            -t "dependabot/dependabot-updater-core:latest" \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ghcr.io/dependabot/dependabot-updater-core \
+            -f Dockerfile.updater-core \
+            .
+
+      - name: Build ecosystem image
+        if: env.DECISION == 'APPROVED:OPEN'
+        env:
+          DOCKER_BUILDKIT: 1
+        run: |
+          docker build \
+            -t "ghcr.io/dependabot/dependabot-updater-${{ matrix.suite.ecosystem }}:$TAG" \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from "$UPDATER_IMAGE" \
+            --cache-from "ghcr.io/dependabot/dependabot-updater-${{ matrix.suite.ecosystem }}" \
+            -f Dockerfile.${{ matrix.suite.name }} \
```

Comment:
Maybe we rename the suite names to be the same as the directory names, e.g. `go_modules`